### PR TITLE
fix: Prevent EPERM/EACCES errors while watching AppMaps if a directory cannot be read

### LINKED ISSUE #880: `glob` fails to read a file or directory and causes the indexer to exit with error code 1
```
Error: EPERM: operation not permitted, scandir 'C:\dev\myproject'
Emitted 'error' event on Glob instance at:
at Glob._readdirError (globalStorage\appland.appmap\node_modules\glob\glob.js:610:14)
at globalStorage\appland.appmap\node_modules\glob\glob.js:553:12
at FSReqCallback.oncomplete (node:fs:188:23) {
errno: -4048,
code: 'EPERM',
syscall: 'scandir',
path: 'C:\\dev\\myproject'
}
```

### PR DESCRIPTION
**Pull request by dustinbyrne**
If the user does not have permissions to read a directory under `--appmap-dir`, an `EPERM`/`EACCES` error would panic the process (even in `--watch` mode).

Instead, these errors will be ignored and we'll continue scanning what we can.

I wasn't sure of an easy way to test an `EPERM` without involving other OS users, so I've omitted the test for now. Behavior should otherwise be the same.

Resolves #880

### CODE REVIEW

File: packages/cli/src/fingerprint/globber.ts [hunk 1]
```diff
@@ -14,7 +15,11 @@ class Globber extends EventEmitter {
   public scanNow() {
     assert(!this.currentGlob);
     this.timeout = undefined;
-    this.currentGlob = new Glob(this.pattern, { ignore: ['**/node_modules/**', '**/.git/**'] })
+    this.currentGlob = new Glob(this.pattern, {
+      ignore: ['**/node_modules/**', '**/.git/**'],
+      strict: false,
```
Comment (apotterri) [inline packages/cli/src/fingerprint/globber.ts:20]: `strict` defaults to `false`, so this has no effect.

**Review by apotterri — CHANGES_REQUESTED** [review]
I think this really needs to include a test.

**Comment by dustinbyrne** [comment]
I've added a test which fails prior to this fix. Turns out it was more simple than expected :+1:

File: packages/cli/src/fingerprint/globber.ts (hunk 1 above)
Comment (apotterri) [reply packages/cli/src/fingerprint/globber.ts:20]: Sorry about that, looks like their doc needs an update: https://github.com/isaacs/node-glob#options .

**Review by apotterri — APPROVED** [review]
Thanks for the test, looks good.

**Comment by apotterri** [comment]
Build's red, though,

**Pull request merged**